compiler/src/dotty/tools/dotc/ast/Trees.scala

```diff
@@ -1245,8 +1249,8 @@ object Trees {
           case Thicket(trees) =>
             val trees1 = transform(trees)
             if (trees1 eq trees) tree else Thicket(trees1)
-          case _ if ctx.reporter.errorsReported =>
-            tree
+          case _ =>
+            transformMoreCases(tree)
```

Comment:
Performance. If we add a rare case, we only override `transformMoreCases`. Before the rare case was in the overridden `transform`. So every match had to first look at the rare cases before it would forward to `super.transform`.